### PR TITLE
[Feature][SQL Transform]Add catalog support for SQL Transform plugin

### PR DESCRIPTION
**Pull request by rewerma**
## Purpose of this pull request

Add catalog support for SQL Transform plugin

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [x] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [x] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [x] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Review by TyrantLucifer — COMMENTED** [review]
Could you please add some unit tests for it?

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-mysql-e2e/src/test/resources/mysqlcdc_to_mysql.conf [hunk 1]
```diff
@@ -37,11 +37,20 @@ source {
...
 transform {
+  sql {
+    source_table_name = "customers_mysql_cdc"
+    query = """ select id, f_binary, f_blob, f_long_varbinary, f_longblob, f_tinyblob, f_varbinary, f_smallint, f_smallint_unsigned, f_mediumint,
+                f_mediumint_unsigned, f_int, f_int_unsigned, f_integer, f_integer_unsigned, f_bigint, f_bigint_unsigned, f_numeric, f_decimal,
+                f_float, f_double, f_double_precision, f_longtext, f_mediumtext, f_text, f_tinytext, f_varchar, f_date, f_datetime, f_timestamp,
+                f_bit1, f_bit64, f_char, f_enum, f_mediumblob, f_long_varchar, f_real, f_time, f_tinyint, f_tinyint_unsigned, f_json, f_year
+                from customers_mysql_cdc """
+    result_table_name = "trans_mysql_cdc"
+  }
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-mysql-e2e/src/test/resources/mysqlcdc_to_mysql.conf:51]: CDC contains INSERT/UPDATE_BEFORE/UPDATE_AFTER/DELETE operations so it is not suitable to use Transform

```suggestion
```

Comment (rewerma) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-mysql-e2e/src/test/resources/mysqlcdc_to_mysql.conf:51]: Y CDC 's operation not suitable to use Transform? I think UPDATA/DELETE operations alsco can use Transform.

Comment (hailin0) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-mysql-e2e/src/test/resources/mysqlcdc_to_mysql.conf:51]: @rewerma 

example cdc data: ROW<id,name>
```
UPDATE_BEFORE  ROW<1, A>
UPDATE_AFTER  ROW<1, B>
```

apply sql transform filter name eq A
```
sql {
    query = "select id, name from where name = 'A' "
}
```


result data：
```
UPDATE_BEFORE  ROW<1, A>
```

how to execute update on sink？

Comment (EricJoy2048) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-mysql-e2e/src/test/resources/mysqlcdc_to_mysql.conf:51]: I think SQL Transform cannot use filter conditions like where in the scenario of CDC synchronization

**Review by hailin0 — APPROVED** [review]

**Review by EricJoy2048 — APPROVED** [review]

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-mysql-e2e/src/test/resources/mysqlcdc_to_mysql.conf (hunk 1 above)
Comment (rewerma) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-mysql-e2e/src/test/resources/mysqlcdc_to_mysql.conf:51]: @EricJoy2048 @hailin0 
I think filter conditions can also be used in the SQL Transform of CDC in certain scenarios, like this: 
```
sql {
    query = "select id, name from t_user where id<>1"
}
```

**Pull request merged**